app/src/test/java/org/fossasia/susi/ai/helper/UtilsTest.kt

```diff
@@ -0,0 +1,18 @@
+package org.fossasia.susi.ai.helper
+
+import junit.framework.Assert.assertEquals
+import org.junit.Test
+
+/**
+ *
+ * Created by arundhati24 on 03/08/2018.
+ */
+class UtilsTest {
+    @Test
+    fun testTruncateEmailAtEnd() {
+        assertEquals("testuser@ ...", Utils.truncateEmailAtEnd("testuser@example.com"))
```

Comment:
Add more emails which may produce errors. Like emails with period and special symbols. Goad of tests is to break the code as much it is to test if it is working correctly

Reply:
What about emails with multiple `@` symbols?

Reply:
@iamareebjamal As this type of email is invalid, hence null should be returned. To check for this, I tried checking for the validity of the email in the ```truncateEmailAtEnd()``` using the ```CredentialsHelper.kt``` class which in turn uses the TextUtils class to check for the validity of the email. But, the tests fail with the error ```java.lang.RuntimeException: Method isEmpty in android.text.TextUtils not mocked. See http://g.co/androidstudio/not-mocked for details```

On checking the link, I found this :

<img width="1130" alt="screen shot 2018-08-10 at 8 04 51 pm" src="https://user-images.githubusercontent.com/30979369/43963880-368c0f50-9cd9-11e8-888e-7ef6b299a3c6.png">

On adding the above code snippet to build.gradle and then running the tests I received an NPE (though I had checked for nullability of email and then used !!)

```
fun truncateEmailAtEnd(email: String?): String? {
        if (!email.isNullOrEmpty()) {
            if (CredentialHelper.isEmailValid(email!!)) {
                val truncateAt = email.indexOf('@')
                if (truncateAt is Int && truncateAt != -1) {
                    return email.substring(0, truncateAt.plus(1)) + " ..."
                }
            }
        }
        return null
    }
```

Reply:
You cannot use android classes in java

Reply:
Would you please suggest how should I account for the cases of multiple '@' or '@' followed by any random symbols?

Reply:
I think that it should simply be truncated after a threshold amount of characters, not according to the position of `@` symbol

Reply:
@iamareebjamal I feel what you have mentioned is true. But, as this is how this has been done on Skills CMS and iOS to anonymise the identity of the user, I need to implement it in the same way in the Android app.
Hence, please suggest what I should do in order to add tests for such a case (although an invalid email is not expected as the response is coming from the server).